internal/errors/errors.go

```diff
@@ -174,4 +174,9 @@ var (
 
 	// As represents errors.As.
 	As = errors.As
+
+	// errNilCustomError represents a function to generate an error that given name's error object is nil.
+	errNilCustomError = func(name string) string {
```

Comment:
Sorry again but kpango has asked if the following suggestion is better.
```suggestion
	errNilCustomError = func(name string) error {
```
@vankichi @hlts2 what do you think??

Reply:
I think both are good for me.
But In terms of how to create an error package, I think it is better to return an error.
and  should we change from unexported value to exported value?

```
errNilCustomError = func(name string) string {
↓
ErrNilCustomError = func(name string) error {
```

Reply:
First of all, I think the name ErrNilCustomError is not clear.
I recommend `ErrExpectedErrorIsNil`

Reply:
@hlts2 I agree with you that it's in error package and it may return an error.
But I think this function should be unexported because it shouldn't be used outside this package.

Reply:
@kevindiu @hlts2 @kpango 
```go
ErrExpectedErrIsNil(n string) error {
    return Errorf("expected err is nil: %s", n)
}
```
How is it?

Reply:
I think it should be `errExpectedErrIsNil` instead of `ErrExpectedErrIsNil` 🤔

Reply:
@kevindiu why do you think?
@hlts2 why do you think it should be exported value? I guessed this will be used only internal errors package.

Reply:
@vankichi 
>
```
ErrExpectedErrIsNil = func(n string) error {
    return Errorf("expected err is nil: %s", n)
}
```How is it?

it looks good to me

Reply:
I think it's getting better, but we need to be concerned about @kevindiu's opinion on whether we should publish this variable or not.

Reply:
@vankichi Like I said, I think this function should be unexported because it shouldn't be used outside this package.
What do you think?

Reply:
@hlts2 to define it is unexported function so that it cannot be used outside this package

Reply:
I got it. thank you.
if this function is not used outside, I think the unexported function is better.

Reply:
so we define `errExpectedErrIsNil` instead of `ErrExpectedErrIsNIl`

Reply:
thank you!